src/main/java/picard/illumina/BasecallsConverter.java

```diff
@@ -0,0 +1,150 @@
+    protected final IlluminaDataProviderFactory factory;
+
+    /**
+     * @param barcodeRecordWriterMap   Map from barcode to CLUSTER_OUTPUT_RECORD writer.  If demultiplex is false, must contain
+     *                                 one writer stored with key=null.
+     * @param demultiplex              If true, output is split by barcode, otherwise all are written to the same output stream.
+     * @param maxReadsInRamPerTile     Configures number of reads each tile will store in RAM before spilling to disk.
+     * @param tmpDirs                  For SortingCollection spilling.
+     * @param numProcessors            Controls number of threads.  If <= 0, the number of threads allocated is
+     *                                 available cores - numProcessors.
+     * @param outputRecordComparator   For sorting output records within a single tile.
+     * @param codecPrototype           For spilling output records to disk.
+     * @param outputRecordClass        Inconveniently needed to create SortingCollections.
+     * @param includeNonPfReads        If true, will include ALL reads (including those which do not have PF set)
+     * @param ignoreUnexpectedBarcodes If true, will ignore reads whose called barcode is not found in barcodeRecordWriterMap,
+     */
+    BasecallsConverter(final Map<String, ? extends ConvertedClusterDataWriter<CLUSTER_OUTPUT_RECORD>> barcodeRecordWriterMap,
+                       final int maxReadsInRamPerTile,
+                       final List<File> tmpDirs,
+                       final SortingCollection.Codec<CLUSTER_OUTPUT_RECORD> codecPrototype,
+                       final boolean ignoreUnexpectedBarcodes,
+                       final boolean demultiplex,
+                       final Comparator<CLUSTER_OUTPUT_RECORD> outputRecordComparator,
+                       final boolean includeNonPfReads,
+                       final BclQualityEvaluationStrategy bclQualityEvaluationStrategy,
+                       final Class<CLUSTER_OUTPUT_RECORD> outputRecordClass,
+                       final int numProcessors,
+                       final IlluminaDataProviderFactory factory) {
+
+        this.barcodeRecordWriterMap = barcodeRecordWriterMap;
+        this.maxReadsInRamPerTile = maxReadsInRamPerTile;
+        this.tmpDirs = tmpDirs;
+        this.codecPrototype = codecPrototype;
+        this.ignoreUnexpectedBarcodes = ignoreUnexpectedBarcodes;
+        this.demultiplex = demultiplex;
+        this.outputRecordComparator = outputRecordComparator;
+        this.includeNonPfReads = includeNonPfReads;
+        this.bclQualityEvaluationStrategy = bclQualityEvaluationStrategy;
+        this.outputRecordClass = outputRecordClass;
+        this.factory = factory;
+
+
+        if (numProcessors == 0) {
+            this.numThreads = Runtime.getRuntime().availableProcessors();
+        } else if (numProcessors < 0) {
+            this.numThreads = Runtime.getRuntime().availableProcessors() + numProcessors;
```

Comment:
Do you want to wrap this in `Math.min(1, …);`, e.g. what if a user passes in -100?